### PR TITLE
Move `tf.keras` integration

### PR DESCRIPTION
**Pull request by toshihikoyanase**
## Motivation

This PR is a part of https://github.com/optuna/optuna/issues/4484. It isolates the `tf.keras` integration.

## Description of the changes

Move `optuna/integration/tfkeras.py` to `optuna_integration/tfkeras.py`

### CODE REVIEW

File: docs/source/reference/index.rst [hunk 1]
```diff
@@ -45,3 +45,12 @@ Keras
...
    optuna.integration.KerasPruningCallback
+
+TensorFlow
+----------
+
+.. autosummary::
+   :toctree: generated/
+   :nosignatures:
+
+   optuna.integration.TFKerasPruningCallback
```
[the remark this answers is not in the document]
Comment (HideakiImamura) [reply docs/source/reference/index.rst:56]: I'm not sure, but if we use optuna-integration, we must install optuna too. So, I think it is fine to use each integration module in optuna namespace.

File: optuna_integration/tfkeras.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+from typing import Any
+from typing import Dict
+from typing import Optional
+import warnings
+
+import optuna
+
+
+with optuna._imports.try_import() as _imports:
```
Comment (HideakiImamura) [inline optuna_integration/tfkeras.py:9]: Could you use `optuna_integration._imports` instead?

Comment (HideakiImamura) [reply optuna_integration/tfkeras.py:9]: Sorry for the confusion. I forgot to note this on the issue https://github.com/optuna/optuna/issues/4484.

File: tests/test_tfkeras.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import numpy as np
+import optuna
+from optuna._imports import try_import
```
Comment (HideakiImamura) [inline tests/test_tfkeras.py:3]: Ditto.

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I have several comments. PTAL.

File: README.md [hunk 4]
```diff
@@ -21,6 +21,7 @@ Optuna-Integration API reference is [here](https://optuna-integration.readthedoc
 * [Chainer](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#chainer)  ([example](https://github.com/optuna/optuna-examples/tree/main/chainer/chainer_integration.py))
 * [ChainerMN](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#chainermn)  ([example](https://github.com/optuna/optuna-examples/tree/main/chainer/chainermn_simple.py))
 * [Keras](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#keras)  ([example](https://github.com/optuna/optuna-examples/tree/main/keras))
+* [tf.keras](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#tensorflow)  ([example](https://github.com/optuna/optuna-examples/tree/main/tfkerastfkeras_integration.py))
```
[the remark this answers is not in the document]
Comment (toshihikoyanase) [reply README.md]: Thanks for your careful review. `/` was removed unexpectedly. Fixed in https://github.com/optuna/optuna-integration/pull/21/commits/caf25a2653fdeb510ce2fff881a540873cf56975 and https://github.com/optuna/optuna-integration/pull/21/commits/c2ad88e29ab26ed48517f218d4ecfa109904ec4f.

File: docs/source/reference/index.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (toshihikoyanase) [reply docs/source/reference/index.rst:56]: Thanks for your reply. In this PR, let us prioritize consistency, and let's discuss it separately.

File: optuna_integration/tfkeras.py (hunk 2 above)
Comment (toshihikoyanase) [reply optuna_integration/tfkeras.py:9]: Ah, https://github.com/optuna/optuna-integration/pull/16 introduced `optuna_integration._imports` yesterday. I'll rebase the master and apply it to this PR.

Comment (toshihikoyanase) [reply optuna_integration/tfkeras.py]: Fixed in https://github.com/optuna/optuna-integration/pull/21/commits/1dd2ebdb8aa183558b66c0f732381fe501e7c2d7.

File: tests/test_tfkeras.py (hunk 3 above)
Comment (toshihikoyanase) [reply tests/test_tfkeras.py]: Fixed in https://github.com/optuna/optuna-integration/pull/21/commits/1dd2ebdb8aa183558b66c0f732381fe501e7c2d7.

**Comment by toshihikoyanase** [comment]
Thank you for your careful review. Fixed the broken link and applied `optuna_integration/_import.py`. PTAL.

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Comment by toshihikoyanase** [comment]
Thanks!

**Pull request merged**